### ISSUE TITLE
Is saving the state by calling .state_dict() sufficient?

### ISSUE THREAD

**Issue by rtxbae** [davda54/sam#73]
```
base_optimizer = torch.optim.Adam
optimizer = SAM(model.parameters(), base_optimizer, lr=0.1)

torch.save({"optz_state_dict":optimizer.state_dict()}, "state.pth")

checkpoint = torch.load("state.pth")
optimizer.load_state_dict(checkpoint["optz_state_dict"])
```

By using the above code, the saved state size is more than halved compared to just purely using pytorch's own `torch.optim.Adam` optimizer. Why such discrepancies, any idea? Resuming them also observed with spike of loss.

The 44mb is using SAM, and the other is Adam:
![image](https://user-images.githubusercontent.com/65499097/176477984-d87f4a6b-a1c2-4921-8123-378e0ff550fe.png)

**Comment by rtxbae** [comment]
Pending

**Comment by DaStapo** [comment]
Wondering the same thing.
